File: 프로그래머스/1/42889. 실패율/실패율.py
def solution(N, stages):
    reacher = len(stages)
    fail_rate_dict = {}
    for i in range(1, N+1):
        challenger = stages.count(i)
        if reacher == 0:
            fail_rate = 0
        else:
            fail_rate = challenger / reacher
        reacher -= challenger
        fail_rate_dict[i] = fail_rate
    
    answer = sorted(fail_rate_dict, key=lambda x: fail_rate_dict[x], reverse=True)
    return answer
        

File: 프로그래머스/1/42889. 실패율/실패율.swift
import Foundation

func solution(_ N:Int, _ stages:[Int]) -> [Int] {
    var challengers = [Int: Int]()
    var failRateDict = [Int: Double]()
    var reachers = stages.count
    
    for stage in stages {
        challengers[stage, default: 0] += 1
    }
    
    for i in 1...N {
        let failRate = Double(challengers[i, default: 0]) / Double(reachers)
        failRateDict[i] = failRate
        reachers -= challengers[i, default: 0]
    }
    
    return failRateDict
                    .sorted { $0.key < $1.key}
                    .sorted { $0.value > $1.value }
                    .map { $0.key }
}